web/api/queries/rrdr.c

```diff
@@ -67,8 +67,8 @@ inline static void rrdr_lock_rrdset(RRDR *r) {
         return;
     }
 
-    rrdset_rdlock(r->st);
     r->has_st_lock = 1;
+    rrdset_rdlock(r->st);
```

Comment:
If we assign `r->has_st_lock = 1` before locking, there is a possibility that unlock will be called when the lock is not held. Wouldn't it be better to introduce a new mutex to guarantee that `has_st_lock` and lock/unlock operations do not enter into a race condition?

Reply:
Also I think the cid 378615 refers to lines https://github.com/netdata/netdata/blob/a82b17ae239ac96973483ea9e861591870bf2cf8/web/api/queries/rrdr.c#L81-L82 ? Those should swap ?

Reply:
I changed it so that it sets/resets while holding a lock